packages/mui-lab/src/internal/pickers/PickersModalDialog.tsx

```diff
@@ -91,6 +92,7 @@ const PickersModalDialog = (props: React.PropsWithChildren<PickersModalDialogPro
     onAccept,
     onClear,
     onDismiss,
```

Comment:
Dialog uses `onDismiss` for it's `onClose` method, ideally it would trigger the `onCancel` so the date is reset